web-common/src/features/metrics-views/workspace/config-parameters/SmallestTimeGrainSelector.svelte

```diff
@@ -54,29 +54,27 @@
     };
   }
 
-  let selectableTimeGrains: TimeGrainOption[] = [];
+  let allowedTimeGrains: TimeGrain[] = [];
   let maxTimeGrainPossibleIndex = 0;
+
+  const timeGrains = Object.values(TIME_GRAIN);
+
   $: if (allTimeRange) {
-    // FIXME: we should be deprecating this getTimeGrainOptions in favor of getAllowedTimeGrains.
-    selectableTimeGrains = getTimeGrainOptions(
+    allowedTimeGrains = getAllowedTimeGrains(
       allTimeRange.start,
       allTimeRange.end
     );
 
-    maxTimeGrainPossibleIndex =
-      selectableTimeGrains.length -
-      1 -
-      selectableTimeGrains
-        .slice()
-        .reverse()
-        .findIndex((grain) => grain.enabled);
+    const maxTimeGrainPossible =
+      allowedTimeGrains[allowedTimeGrains.length - 1];
+    maxTimeGrainPossibleIndex = timeGrains
+      .map((grain) => grain.label)
+      .indexOf(maxTimeGrainPossible.label);
```

Comment:
You can combine these two calls into one loop with `findIndexOf(grain => grain.label === maxTimeGrainPossible.label)`, fyi. Beyond only doing 1 loop, it will also early exit when it finds a match